packages/kas/src/nodes.ts

```diff
@@ -162,65 +212,54 @@ class Expr {
 
     // returns the name of this expression's constructor as a string
     // only used for testing and debugging (the ugly regex is for IE8)
-    name() {
-        if (this.func.name) {
-            return this.func.name;
-        } else {
-            return this.func.toString().match(/^function\s*([^\s(]+)/)[1];
-        }
+    name(): string {
+        return this.func.name;
     }
 
     // returns a string representing current node structure
-    repr() {
+    repr(): string {
         return (
             this.name() +
             "(" +
             _.map(this.args(), function (arg) {
-                return _.isString(arg) ? arg : arg.repr();
+                return _.isString(arg) || _.isNumber(arg) ? arg : arg.repr();
             }).join(",") +
             ")"
         );
     }
 
     // removes all negative signs
-    strip() {
+    strip(): Expr {
         return this.recurse("strip");
     }
 
     // canonically reorders all commutative elements
-    normalize() {
+    normalize(): Expr {
         return this.recurse("normalize");
     }
 
     // expands the expression
-    expand() {
+    expand(options?): Expr {
         return this.recurse("expand");
     }
 
     // naively factors out like terms
-    factor(options) {
+    factor(options): Expr {
         return this.recurse("factor", options);
     }
 
     // collect all like terms
-    collect(options) {
+    collect(options?: Options): Expr {
         return this.recurse("collect", options);
     }
 
     // strict syntactic equality check
-    equals(other) {
+    equals(other: Expr): boolean {
         return this.normalize().print() === other.normalize().print();
     }
 
     // expand and collect until the expression no longer changes
-    simplify(options) {
-        options = _.extend(
-            {
-                once: false,
-            },
-            options,
-        );
-
+    simplify(options: Options = {once: false}) {
```

Comment:
This seems like a logic change, was it intentional? In the original one, if you passed `simplify({hello: "world"})` options would be `{once: false, hello: "world"}`; in the updated one it would be `{hello: "world"}`. I'm sure you checked, just wanted to be double sure.